modules/markup/html.go

```diff
@@ -377,8 +377,17 @@ func (ctx *postProcessCtx) visitNode(node *html.Node, visitText bool) {
 
 					lnk := string(link)
 					lnk = util.URLJoin(prefix, lnk)
+
 					link = []byte(lnk)
 				}
+				if setting.CamoServerURL != "" {
+					lnk := string(link)
+					lnkURL, _ := url.Parse(lnk)
+					if lnkURL.IsAbs() && !strings.HasPrefix(lnk, setting.AppURL) {
```

Comment:
While Go-Camo (and other similar image/media proxies) will indeed support proxying HTTPS images, for performance reasons you may choose to filter HTTPS requests out from proxying, and let the client simply fetch those as they are. 

An example that would avoid proxying https images and only secure proxy http.
```
&& ! strings.HasPrefix(lnk, "https:")
```

Reply:
I think we can add this to settings (http / always ...)